src/frontend/src/pages/SettingsPage/UserSecureSettings/UserSecureSettings.tsx

```diff
@@ -82,17 +88,17 @@ const UserSecureSettings: React.FC<SecureSettingsProps> = ({ currentSettings })
               Save
             </NERSuccessButton>
           </Box>
-        )
+        ))
       }
     >
       <Grid container>
         {!edit ? (
           <UserSecureSettingsView settings={currentSettings} />
         ) : (
-          <UserSecureSettingsEdit currentSettings={currentSettings} onSubmit={handleConfirm} />
+          !isGuest(user.role) && <UserSecureSettingsEdit currentSettings={currentSettings} onSubmit={handleConfirm} />
         )}
       </Grid>
-      {edit && (
+      {!isGuest(user.role) && edit && (
```

Comment:
you don't need this check or the check directly above. If a user can't click the edit button, then they won't be able to see either of these views anyways

Reply:
got it. i was having issues with the edit and cancel button rendering if the user was a guest, so i added the conditions, but this works.